java-runtime-builder-app/src/main/java/com/google/cloud/runtimes/builder/buildsteps/SourceBuildRuntimeImageBuildStep.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.runtimes.builder.buildsteps;
+
+import com.google.cloud.runtimes.builder.config.domain.BuildContext;
+import com.google.cloud.runtimes.builder.config.domain.JdkServerLookup;
+import com.google.inject.Inject;
+
+public class SourceBuildRuntimeImageBuildStep extends RuntimeImageBuildStep {
+
+  @Inject
+  SourceBuildRuntimeImageBuildStep(JdkServerLookup jdkServerLookup) {
+    super(jdkServerLookup);
+  }
+
+  @Override
+  protected String getArtifact(BuildContext buildContext) {
+    String providedArtifactPath = buildContext.getRuntimeConfig().getArtifact();
+
+    // Require that the artifact name is explicitly provided by the user.
+    if (providedArtifactPath == null) {
+      throw new IllegalStateException("Unable to determine the artifact path. In order to build "
+          + "from source, the path to the artifact must be specified in the runtime_config.artifact"
```

Comment:
I was trying to avoid references to app.yaml, since technically the deployment descriptor can be named `*.yaml` - do you think it's better to use app.yaml?